### DISCUSSION TITLE
problem when trying to personalize cost function

### DISCUSSION THREAD

**Discussion by rmoumni15** [awslabs/gluonts#1444] [Q&A]
Hello,

I am trying to implement my own cost function to train my model a multi-layer perceptron, my concern is that during training I receive the following problem :

>  Check failed: !AGInfo::IsNone(*i): Cannot differentiate node because it is not in a computational graph. You need to set is_recording to true or use autograd.record() to save computational graphs for backward. If you want to differentiate the same graph twice, you need to pass retain_graph=True to backward.

i tried :
 
  ```
def myloss(forecast, target):
        .
        .
        .
        .     
        with autograd.record():
             loss= nd.array(np.sqrt((r-1)**2+(beta-1)**2+(gamma-1)**2))
        return loss
```
and in `hybrid_forward` i call my lost function 

  ```
def hybrid_forward(self, F, past_target, future_target):
        prediction = self.nn(past_target)     
        loss = ( 1 - self.myloss( prediction,future_target) ).mean(axis=-1)
        return loss
```
I also noticed that when I calculate a loss without converting my parameters from mxnet ndarray to numpy ndarray it works, but in my case it is mandatory since I have to perform several intermediate calculations.
